scripts/ci/cmake.sh

```diff
@@ -6,119 +6,128 @@
 source $(realpath $(dirname $0))/common.sh
 
 die_syntax() {
-  echo "Syntax: $0 -s SRC_DIR -b BLD_DIR -m MLIR_DIR [-i INST_DIR] [-t (Release|Debug|RelWithDebInfo)] [-c (clang|gcc)] [-g (gcc toolchain)] [-l (ld|lld|gold|mold)] [-S] [-n N]"
+  echo "Syntax: $0 -s SRC_DIR -b BLD_DIR -m MLIR_DIR [-i INST_DIR]"
+  echo "          [-t (Release|Debug|RelWithDebInfo)] [-c (clang|gcc)] [-g (gcc toolchain)]"
+  echo "          [-l (ld|lld|gold|mold)] [-R] [-S] [-n N]"
   echo ""
-  echo "  -i: Optional install dir, default to system"
-  echo "  -t: Optional build type flag, default to Release"
-  echo "  -c: Optional compiler flag, default to clang"
+  echo "  -i: Optional install dir, defaults to system"
+  echo "  -t: Optional build type flag, defaults to Release"
+  echo "  -c: Optional compiler flag, defaults to clang"
   echo "  -g: Optional gcc toolchain flag, may be needed by clang"
-  echo "  -l: Optional linker flag, default to system linker"
-  echo "  -S: Optional sanitizer flag, default to none"
-  echo "  -G: Optional GPU support flag, default to none"
-  echo "  -n: Optional link jobs flag, default same as CPUs"
+  echo "  -l: Optional linker flag, defaults to system linker"
+  echo "  -R: Optional request to remove BLD_DIR before CMake"
+  echo "  -S: Optional sanitizer flag, defaults to none"
+  echo "  -G: Optional GPU support flag, defaults to none"
+  echo "  -n: Optional link job flag, defaults to nproc"
   exit 1
 }
 
 # Cmd-line opts
-while getopts "s:b:i:m:t:c:g:l:n:SG" arg; do
-  case ${arg} in
-    s)
-      SRC_DIR=$(realpath ${OPTARG})
+while test $# -gt 0; do
```

Comment:
Why the move from `getopts`?

Reply:
To support the previous command line interface like `-G` without arguments. Now, `-G` can also take `cuda` or `vulkan` (or anything else we may want later). This could not be expressed with `getopts` like checking if the "next" argument is a flag or a value.

Reply:
Ok, I will bring back `getopt`. I thought people are using these scripts and wanted to provide "no change".